src/error_ext.rs

```diff
@@ -0,0 +1,107 @@
+use Fail;
+use std::error::Error;
+use std::fmt::{self, Debug, Display};
+use std::sync::Mutex;
+
+/// Wrapper for std errors to make them Sync.
+///
+/// This exists to coerce existing types that are only `Error + Send + 'static`
+/// into a `Fail`-compatible representation.
+///
+/// Unfortunately, this requires wrapping the error in a `Mutex`, which must be
+/// locked for every `Debug`/`Display`.
+pub struct SyncErr<T>(Mutex<T>);
```

Comment:
nits:

* Could you make this a `{ }` struct? I prefer this for items in the public API.
* I'd prefer `SyncFailure` for the name, I think.

Reply:
Should it be `SyncFailure` go go with the crate name, or `SyncFail` go go with the existing `Fail` trait?

Reply:
The idea of the naming convention is that traits are verbs, types are nouns. See `Read` and `BufReader` for example.

Reply:
Ah, gotcha! Makes sense.